### ISSUE TITLE
Get familiar with Gene Page feature

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#833]
Looking through the Fall 2017 BioDB wiki, there isn't a good page to point to.  However, on the GRNsight wiki, we do have a documentation page here:

https://github.com/dondi/GRNsight/wiki/Web-API-Guide

@Kevboe should look over this documentation and see if it is correct and make adjustments as necessary.

**Comment by kdahlquist** [comment]
@kdahlquist and @Kevboe met on Friday to go over the gene page feature.
- species and taxon ID are already stored in GRNstate
- there is already a method to retrieve the species and taxon ID from GRNstate to populate queries, 
- probably need to write some tests to validate both of the above #682 
- then need to create the curated list of species/taxon IDs that we are going to start out with, issue #721 
- need two methods for user to indicate which species they want (from the curated list)
    - one method is to encode the information and read it from a GRNmap input workbook, #720
    - another method is to ask the user directly.  We discussed having a collapsible section where it has a drop-down menu listing the choices of species that GRNsight supports.

**Comment by kdahlquist** [comment]
@Kevboe will do a verify and fill in the blank on steps 1, 2, 3, @dondi thinks that John went as far as step 3, but this should be verified. That code might be on John's branch.  If anything is missing from step 1-3, @Kevboe should fill it in.

Tests to write are to make sure that when the GRNsight code issues gene API queries that the queries include the species and/or taxon ID as needed.

**Comment by Kevboe** [comment]
I was able to confirm that the species and taxon ID are stored in GRNstate, with an existing function for when data is read in from an .xml file. There is no explicit method, but the species and taxon can be accessed by calling GRNstate.species or GRNstate.taxon respectively. I wrote a small test in grnstate-tests that currently tests that the GRNstate.species and GRNstate.taxon are accessible and correctly  set in the grnState class. I created a dictionary of species/taxon IDs from the list provided on issue #721 . The current system is a key value pair of the English name and a tuple of the species and taxon ID. This list can be expanded and used in the future to update the GRNstate variables.  Looking over the API-tests it felt that it was the wrong place to test that the queries included a the species and taxonID as we have hard coded them for the tests the API calls and information format. The API calls also currently filter the passed in query of symbol, species, and taxon for only what they need. This functionality may need to be changed as the query changes to handle other species.  Given other homework and tests in other classes I felt as if this week may not have been as productive as I would've liked, but I was still able to grow my understanding of the code and expand my notes. If you would like me to type them for posterity's sake, I can as part of next weeks work.

**Comment by Kevboe** [comment]
An idea that I had for the input of species was a detection system that will autofill the drop down menu if a species is identified, but will otherwise prompt the user for input if it is not detected. This way if there was an issue in the detection the user can also manually see and then fix it before they submit the workbook.

**Comment by Kevboe** [comment]
Worked on #5 on the list, will write up more on what I did on issue #720.  I set up an 'identified' variable that can be passed between the manual input and the read in input. The list itself needs to be created, I was planning on using a bootstrap drop down list, if there isn't any issues with that then I will proceed with it this week. I plan to have an alert to let the user know what was detected and select the right one  on the list, and allow them to adjust if there was a mistake. I might need help making sure the list can interact with variables properly and self update if detected, as well as writing to tests to make sure that I can prove that it works.
